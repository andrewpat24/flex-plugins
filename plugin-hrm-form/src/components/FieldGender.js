import React from 'react';
import { ErrorText,
         StyledLabel,
         StyledMenuItem,
         StyledSelect,
         TextField
} from '../Styles/HrmStyles';

{/* <TextField>
<StyledLabel htmlFor="ChildInformation_Gender">Gender</StyledLabel>
<StyledSelect 
  name="gender"
  id="ChildInformation_Gender"
  value={this.props.form.childInformation.gender}
  onChange={(e) => this.props.handleChange(taskId, ['childInformation'], e)}>
  <StyledMenuItem hidden selected disabled value="">Select</StyledMenuItem>
  <StyledMenuItem value="Male">Male</StyledMenuItem>
  <StyledMenuItem value="Female">Female</StyledMenuItem>
  <StyledMenuItem value="Other">Other</StyledMenuItem>
  <StyledMenuItem value="Unknown">Unknown</StyledMenuItem>
</StyledSelect>
</TextField> */}

const FieldGender = (props) => {
  return (
    <TextField>
      <StyledLabel htmlFor="ChildInformation_Gender">Gender<span style={{color: 'red'}}>*</span></StyledLabel>
      <StyledSelect
        error={props.form.childInformation.gender.error !== null}
        name='gender'
        id="ChildInformation_Gender"
        value={props.form.childInformation.gender.value}
        onBlur={() => props.handleBlur()}
        onChange={(e) =>
          props.handleChange(props.taskId,
                             ['childInformation'],
                             e)}
        onFocus={() => 
          props.handleFocus(props.taskId,
                            ['childInformation'],
                            'gender')}
      >
        <StyledMenuItem value="Male">Male</StyledMenuItem>
        <StyledMenuItem value="Female">Female</StyledMenuItem>
        <StyledMenuItem value="Other">Other</StyledMenuItem>
        <StyledMenuItem value="Unknown">Unknown</StyledMenuItem>
      </StyledSelect>
      {props.form.childInformation.gender.error ?
        <ErrorText>{props.form.childInformation.gender.error}</ErrorText> :
        ''
      }
    </TextField>
  );
};

export default FieldGender;
