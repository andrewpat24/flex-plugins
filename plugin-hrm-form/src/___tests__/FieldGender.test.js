jest.mock('../Styles/HrmStyles', () => {
  return {
    ErrorText: 'ErrorText',
    StyledLabel: 'StyledLabel',
    StyledMenuItem: 'StyledMenuItem',
    StyledSelect: 'StyledSelect',
    TextField: 'TextField'
  };
});
import React from 'react';
import FieldGender from '../components/FieldGender';
import renderer from 'react-test-renderer';

test('FieldGender renders initially', () => {
  const form = {
    childInformation: {
      gender: {
        value: '',
        touched: false,
        error: null
      }
    }
  };
  const handleChange = jest.fn();
  const taskId = '';
  const component = renderer.create(
    <FieldGender
      form={form}
      handleChange={handleChange}
      taskId={taskId}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('FieldGender renders when valid', () => {
  const form = {
    childInformation: {
      gender: {
        value: 'Female',
        touched: false,
        error: null
      }
    }
  };
  const handleChange = jest.fn();
  const taskId = '';
  const component = renderer.create(
    <FieldGender
      form={form}
      handleChange={handleChange}
      taskId={taskId}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('FieldGender renders errors when not valid', () => {
  const form = {
    childInformation: {
      gender: {
        value: '',
        touched: true,
        error: 'This field is required'
      }
    }
  };
  const handleChange = jest.fn();
  const taskId = '';
  const component = renderer.create(
    <FieldGender
      form={form}
      handleChange={handleChange}
      taskId={taskId}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
