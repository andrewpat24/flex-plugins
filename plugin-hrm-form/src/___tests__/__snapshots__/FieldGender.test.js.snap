// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldGender renders errors when not valid 1`] = `
<TextField>
  <StyledLabel
    htmlFor="ChildInformation_Gender"
  >
    Gender
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      *
    </span>
  </StyledLabel>
  <StyledSelect
    error={true}
    id="ChildInformation_Gender"
    name="gender"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    value=""
  >
    <StyledMenuItem
      value="Male"
    >
      Male
    </StyledMenuItem>
    <StyledMenuItem
      value="Female"
    >
      Female
    </StyledMenuItem>
    <StyledMenuItem
      value="Other"
    >
      Other
    </StyledMenuItem>
    <StyledMenuItem
      value="Unknown"
    >
      Unknown
    </StyledMenuItem>
  </StyledSelect>
  <ErrorText>
    This field is required
  </ErrorText>
</TextField>
`;

exports[`FieldGender renders initially 1`] = `
<TextField>
  <StyledLabel
    htmlFor="ChildInformation_Gender"
  >
    Gender
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      *
    </span>
  </StyledLabel>
  <StyledSelect
    error={false}
    id="ChildInformation_Gender"
    name="gender"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    value=""
  >
    <StyledMenuItem
      value="Male"
    >
      Male
    </StyledMenuItem>
    <StyledMenuItem
      value="Female"
    >
      Female
    </StyledMenuItem>
    <StyledMenuItem
      value="Other"
    >
      Other
    </StyledMenuItem>
    <StyledMenuItem
      value="Unknown"
    >
      Unknown
    </StyledMenuItem>
  </StyledSelect>
  
</TextField>
`;

exports[`FieldGender renders when valid 1`] = `
<TextField>
  <StyledLabel
    htmlFor="ChildInformation_Gender"
  >
    Gender
    <span
      style={
        Object {
          "color": "red",
        }
      }
    >
      *
    </span>
  </StyledLabel>
  <StyledSelect
    error={false}
    id="ChildInformation_Gender"
    name="gender"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    value="Female"
  >
    <StyledMenuItem
      value="Male"
    >
      Male
    </StyledMenuItem>
    <StyledMenuItem
      value="Female"
    >
      Female
    </StyledMenuItem>
    <StyledMenuItem
      value="Other"
    >
      Other
    </StyledMenuItem>
    <StyledMenuItem
      value="Unknown"
    >
      Unknown
    </StyledMenuItem>
  </StyledSelect>
  
</TextField>
`;
